posts/2020-10-05-Rust-1.47.md

```diff
@@ -0,0 +1,230 @@
+   2: std::sys_common::backtrace::_print_fmt
+             at src/libstd/sys_common/backtrace.rs:78
+   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
+             at src/libstd/sys_common/backtrace.rs:59
+   4: core::fmt::write
+             at src/libcore/fmt/mod.rs:1076
+   5: std::io::Write::write_fmt
+             at src/libstd/io/mod.rs:1537
+   6: std::sys_common::backtrace::_print
+             at src/libstd/sys_common/backtrace.rs:62
+   7: std::sys_common::backtrace::print
+             at src/libstd/sys_common/backtrace.rs:49
+   8: std::panicking::default_hook::{{closure}}
+             at src/libstd/panicking.rs:198
+   9: std::panicking::default_hook
+             at src/libstd/panicking.rs:217
+  10: std::panicking::rust_panic_with_hook
+             at src/libstd/panicking.rs:526
+  11: std::panicking::begin_panic
+             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9/src/libstd/panicking.rs:456
+  12: playground::main
+             at src/main.rs:2
+  13: std::rt::lang_start::{{closure}}
+             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9/src/libstd/rt.rs:67
+  14: std::rt::lang_start_internal::{{closure}}
+             at src/libstd/rt.rs:52
+  15: std::panicking::try::do_call
+             at src/libstd/panicking.rs:348
+  16: std::panicking::try
+             at src/libstd/panicking.rs:325
+  17: std::panic::catch_unwind
+             at src/libstd/panic.rs:394
+  18: std::rt::lang_start_internal
+             at src/libstd/rt.rs:51
+  19: std::rt::lang_start
+             at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9/src/libstd/rt.rs:67
+  20: main
+  21: __libc_start_main
+  22: _start
+```
+
+Now, in Rust 1.47.0, you'll see this instead:
+
+```text
+thread 'main' panicked at 'explicit panic', src/main.rs:2:5
+stack backtrace:
+   0: std::panicking::begin_panic
+             at /rustc/d6646f64790018719caebeafd352a92adfa1d75a/library/std/src/panicking.rs:497
+   1: playground::main
+             at ./src/main.rs:2
+   2: core::ops::function::FnOnce::call_once
+             at /rustc/d6646f64790018719caebeafd352a92adfa1d75a/library/core/src/ops/function.rs:227
+```
+
+This makes it much easier to see where the panic actually originated, and
+you can still set `RUST_BACKTRACE=full` if you want to see everything.
+
+#### LLVM 11
+
+We have [upgraded to LLVM 11](https://github.com/rust-lang/rust/pull/73526/).
+The compiler still supports being compiled with LLVM versions as old as 8,
+but by default, 11 is what you'll be getting.
+
+#### Control Flow Guard on Windows
+
+`rustc` [now supports](https://github.com/rust-lang/rust/pull/73893/) `-C
+control-flow-guard`, an option that will turn on [Control Flow
+Guard](https://docs.microsoft.com/en-us/windows/win32/secbp/control-flow-guard)
+on Windows. Other platforms ignore this flag.
+
+### Library changes
+
+[Rustdoc has gained support for the Ayu theme](https://github.com/rust-lang/rust/pull/71237/).
```

Comment:
That feels weird to call out as a library change.

Reply:
yeah, this is rustdoc changes. Some recent crates on docs.rs have the ayu theme, but not all
since docs.rs doesn't rebuilt old versions.

Reply:
Where do you think the right place is? It sorta doesn't fit into the language/stdlib split we use.

Reply:
I personally just wouldn't include it, it doesn't feel major enough for blog to me. But I'm not a very color-y person :)

Perhaps just moving it to the end so it's not really in the library changes section so prominently?